### PR TITLE
Use websocket v2

### PR DESCRIPTION
**Pull request by manuquentin**

### CODE REVIEW

File: src/websocket-client.js [hunk 1]
```diff
@@ -64,8 +64,8 @@ class WebSocketClient extends Emitter {
 
       if (!this.initialized) {
         this.handleMessage(message, this.socket);
-      } else {
-        this.eventEmitter.emit(message.name, message);
+      } else if (message.op === 'event') {
+        this.eventEmitter.emit(message.name || message.data.name, message.data || message);
```
Comment (sileht) [inline src/websocket-client.js]: Should we support V1 and V2 in parallel ? if yes:

Passing `&version=2` is telling the server the maximum version supported. It's up to the server to choice the version that both end must use.

When you receive the first `init` message, you have the version that both end must use.

You can take a look here:
https://github.com/TinxHQ/nestbox-ui-ng/pull/213/files#diff-15613bc5845576e1c2bfdef7efdcaf7eR42-R68

File: src/websocket-client.js [hunk 2]
```diff
@@ -64,8 +68,8 @@ class WebSocketClient extends Emitter {
 
       if (!this.initialized) {
         this.handleMessage(message, this.socket);
-      } else {
```
Comment (sileht) [inline src/websocket-client.js]: If it's version 1, you must keep this else.

File: src/websocket-client.js [hunk 3]
```diff
@@ -120,13 +124,18 @@ class WebSocketClient extends Emitter {
     this.token = token;
 
     if (this.socket) {
+      // If still connected, send the token to the WS
+      if (this.socket.readyState === this.socket.OPEN) {
```
Comment (sileht) [inline src/websocket-client.js]: You can send it only version >= 2

Comment (manuquentin) [reply src/websocket-client.js]: Done

File: src/websocket-client.js (hunk 1 above)
Comment (manuquentin) [reply src/websocket-client.js]: Yes but I prefer let the user choose the version he want to use.

**Pull request merged**